samcli/commands/local/lib/local_lambda.py

```diff
@@ -86,6 +88,33 @@ def is_debugging(self):
         """
         return bool(self.debug_context)
 
+    def _sanitize_inlinecode(self, inlinecode):
+        """
+        Verify if InlineCode needs runtime substitution
+        """
+        # InlineCode isn't type safe
+        if isinstance(inlinecode, dict):
+            # We expect to have only 1 keypair..
+            inlinecode = list(inlinecode.values())[0]
+        # If ${something} found inside the code, raise a NotImplemented error
+        if re.search(r'\${[^!]\S+}', inlinecode):
+            LOG.debug('Not able to parse inline code: %s', inlinecode)
+            raise NotImplementedError('Not able to parse inline code because of !Sub presence')
+        return inlinecode
+
+    # FIXME: randommize dump path and make sure not to overwrite an existing file
+    def _dump_inlinecode(self, inlinecode, path):
+        """
+        Dumps InlineCode's Function string to a file and register a function to remove the file when processes finish
+        """
+
+        inlinecode = self._sanitize_inlinecode(inlinecode)
+        file_location = '{}/index'.format(path)
+        with open(file_location, 'w+') as f:
+            f.write(inlinecode)
+        atexit.register(lambda: os.remove(file_location))
```

Comment:
We want this to be cleaned up after the invoke not after the process exists. For `start-lambda` and `start-api`, these files will hang around for a longer period of time for no reason. We should clean them up after invoking. This can be done with a context manager instead.

Reply:
got it, going to revise heavily if taking the provider path